client/components/ChapterViewComponent.js

```diff
@@ -1,60 +1,123 @@
-      imageComponent = <img src={imagePath} {...imageProps} />
-      break
+  updateSlider (updatedValue, totalPages) {
+    if (updatedValue >= 1 && updatedValue <= totalPages) {
+      this.setState({ slider: updatedValue })
+    }
   }
 
-  return (
-    <div>
-      <AppBar
-        title={title}
-        iconElementLeft={<IconButton onClick={back}><NavigationArrowBack /></IconButton>}
-        iconElementRight={<IconButton onClick={prevClicked}><ContentUndo /></IconButton>}
-        style={{ position: 'fixed' }}
+  render () {
+    const specificManga = this.props.manga.get(this.props.mangaName)
+    const chapter = specificManga.get('chapters').get(this.props.chapterNum)
+    const type = `http://${specificManga.get('type')}`
+    const title = `${specificManga.get('title')} - ${chapter.get('name')}`
+    const currPage = chapter.get('currentPage')
+
+    const onlineURL = chapter.get('pages').get(currPage)
+    const totalPages = chapter.get('pages').count()
+    const downloadState = chapter.get('download').get('state')
+
+    const imageClicked = (xOffset, yOffset, dimensions) => {
+      // If click was on the right half of the image then go to the next page,
+      // otherwise go to the previous page.
+      if (xOffset > dimensions.width / 2) {
+        this.updateSlider(this.state.slider + 1, totalPages)
+        this.props.update(specificManga, this.props.chapterNum, 1)
+      } else {
+        this.updateSlider(this.state.slider - 1, totalPages)
+        this.props.update(specificManga, this.props.chapterNum, -1)
+      }
+    }
+
+    const dropDownClicked = () => this.setState({ navigationVisible: !this.state.navigationVisible })
+
+    const sliderChanged = (value) => {
+      this.updateSlider(value, totalPages)
+      const diff = value - currPage - 1
+
+      if (diff !== 0) {
+        this.props.update(specificManga, this.props.chapterNum, diff)
+      }
+    }
+
+    let imagePath = null
+    let downloaded = false
+    if (downloadState === DOWNLOADING || downloadState === NOT_DOWNLOADED) {
+      imagePath = onlineURL
+    } else if (downloadState === DOWNLOADED) {
+      imagePath = 'manga://' + path.join(
+        specificManga.get('name'),
+        this.props.chapterNum + '',
+        encodeURIComponent(onlineURL)
+      )
+      downloaded = true
+    } else {
+      throw new Error('Invalid download state')
+    }
+
+    const imageComponent = (
+      <ImageComponent
+        src={imagePath}
+        type={type}
+        style={{ width: '100%' }}
+        downloaded={downloaded}
+        onImageClick={imageClicked}
       />
-      <br /><br /><br /><br />
-      {imageComponent}
-    </div>
-  )
+    )
+
+    let sliderComponent = <div />
+    if (this.state.navigationVisible) {
+      sliderComponent = (
+        <SliderComponent
+          currValue={this.state.slider}
+          totalPages={totalPages}
+          onSliderChanged={sliderChanged}
+        />
+      )
+    }
+
+    return (
+      <div>
```

Comment:
Is it standard to return the hard coded tags or is it better to use `document.createElement("div")`?